### PR TITLE
Pa02

### PR DESCRIPTION
**Pull request by ErikRulle**
Håvard Molversmyr and Erik Rullestad

### CODE REVIEW

**Review by molversmyr — APPROVED** [review]
Approved!!

File: src/erik_rullestad_ex/pa02/chutes_simulations.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    default_extra_steps = 1
+
+    def __init__(self, board, extra_steps=None):
+        super().__init__(board)
+        self.extra_steps = extra_steps
+        if self.extra_steps is None:
+            self.extra_steps = self.default_extra_steps
+
+    def move(self):
+        if self.get_position() in self.board.chutes.values():
```
Comment (huynhngoc) [inline src/erik_rullestad_ex/pa02/chutes_simulations.py:67]: Nice thinking. But the player doesn't need to be bitten by a snake to be at the tail of a snake.

File: src/erik_rullestad_ex/pa02/chutes_simulations.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+            self.n_steps += 1
+        else:
+            super().move()
+
+    def get_position(self):
+        return self.position
+
+    def get_steps(self):
+        return self.n_steps
```
Comment (huynhngoc) [inline src/erik_rullestad_ex/pa02/chutes_simulations.py:81]: You don't need to rewrite these functions, they are the same with the parent so they should have been inherited from the parent class.

File: src/erik_rullestad_ex/pa02/chutes_simulations.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    default_dropped_steps = 1
+
+    def __init__(self, board, dropped_steps=None):
+        super().__init__(board)
+        self.dropped_steps = dropped_steps
+        if self.dropped_steps is None:
+            self.dropped_steps = self.default_dropped_steps
+
+    def move(self):
+        if self.get_position() in self.board.ladders.values():
```
Comment (huynhngoc) [inline src/erik_rullestad_ex/pa02/chutes_simulations.py:94]: Also, there are still cases that the player doesn't climb the ladder to be at the top of a ladder.
The right way to do this is to create a attribute in this class as a "flag" to check if a player made a special move in the last round.

File: src/erik_rullestad_ex/pa02/chutes_simulations.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                pass
+        else:
+            super().move()
+
+    def get_position(self):
+        return self.position
+
+    def get_steps(self):
+        return self.n_steps
+
```
Comment (huynhngoc) [inline src/erik_rullestad_ex/pa02/chutes_simulations.py:114]: Again, remove these.

File: src/erik_rullestad_ex/pa02/chutes_simulations.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    def __init__(self, player_field=None, board=None,
+                 seed=None, randomize_players=True):
+        self.player_field = player_field
+        self.board = board
+        if self.board is None:
+            self.board = Board()
+        self.seed = seed
+        self.randomize_players = randomize_players
+        if self.randomize_players is True:
+            random.shuffle(self.player_field)
```
Comment (huynhngoc) [inline src/erik_rullestad_ex/pa02/chutes_simulations.py:126]: This should be done at the beginning of every single_game

File: src/erik_rullestad_ex/pa02/test_pa02.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+        p = cs.Player(b)
+        p.move()
+        assert p.get_position() is p.position
+
+    def test_get_steps(self):
+        b = cs.Board()
+        p = cs.Player(b)
+        p.move()
+        p.move()
+        assert p.get_steps() is 2
```
Comment (huynhngoc) [inline src/erik_rullestad_ex/pa02/test_pa02.py:54]: `2` is of type `int`, which is a primitive type, and doesn't need to use `is` for comparison. `==` is more than enough here.

**Review by huynhngoc — APPROVED** [review]
Nice. Approved.

Please note that docstring is a very important part of your code, and you may fail the Jan/Jun block project if missing them.

**Pull request merged**